bluetooth/src/commonMain/kotlin/device/DeviceConnectionManager.kt

```diff
@@ -64,82 +88,59 @@ abstract class BaseDeviceConnectionManager(
     abstract suspend fun performAction(action: DeviceAction)
 
     suspend fun handleNewRssi(rssi: Int) {
-        stateRepo.takeAndChangeState {
-            it.rssiDidUpdate(rssi)
+        _rssiUpdate.emit(rssi)
+    }
+
+    fun handleNewMtu(mtu: Int) {
+        _mtu.value = mtu
+    }
+
+    fun startConnecting() {
+        launch {
```

Comment:
Will probably also need to dig more into this with the next PRs, but we should consider `tryEmit` + warning if it fails. 

In general I am a bit worried about ordering of all these launches and using different SharedFlows e.g. for start and cancel connect.. should this not be one SharedFlow with an `enum class` or something?